src/starlite_saqlalchemy/dto.py

```diff
@@ -72,9 +84,46 @@ class Attrib(NamedTuple):
     """If provided, used for the pydantic model for this attribute."""
 
 
+class BaseConfig(BaseConfig_):
+    """Base config for generated pydantic models."""
+
+    orm_mode = True
+
+
+class MapperBind(BaseModel):
+    """Produce an SQLAlchemy instance with values from a pydantic model."""
+
+    __sqla_model__: type[DeclarativeBase]
+
+    class Config(BaseConfig):
+        """Config for MapperBind pydantic models."""
+
+    def __init_subclass__(cls, model: type[DeclarativeBase]) -> None:
+        cls.__sqla_model__ = model
+        return super().__init_subclass__()
+
+    def mapper(self) -> DeclarativeBase:
+        """Fill the binded SQLAlchemy model recursively with values from this
+        dataclass."""
+        as_model = {}
+        for field in self.__fields__.values():
+            value = getattr(self, field.name)
+            if isinstance(value, (list, tuple)):
+                value = [el.mapper() if isinstance(el, MapperBind) else el for el in value]
+            if isinstance(value, MapperBind):
+                value = value.mapper()
+            as_model[field.name] = value
+        return self.__sqla_model__(**as_model)
+
+
 def _construct_field_info(elem: Column | RelationshipProperty, purpose: Purpose) -> FieldInfo:
     default = getattr(elem, "default", None)
-    if purpose is Purpose.READ or default is None:
+    nullable = getattr(elem, "nullable", False)
+    if purpose is Purpose.READ:
+        return FieldInfo(...)
+    if default is None:
+        if not nullable:
+            return FieldInfo(default=None)
```

Comment:
Is this correct? If default is `None` and column is not nullable, should it be ellipsis?

So this is distinguishing between an explicit `None` default and the attribute not existing, right? Couldn't hurt to use a sentinel value other than `None` in the `getattr()` call also.

Reply:
You're right, I found that bug too just after opening this PR, forgot to push.